pkg/cmd/workflow/run/run.go

```diff
@@ -0,0 +1,382 @@
+
+			# Run the workflow file 'triage.yml' with command line inputs
+			$ gh workflow run triage.yml -- --name=scully --greeting=hello
+
+			# Run the workflow file 'triage.yml' with JSON via STDIN
+			$ echo '{"name":"scully", "greeting":"hello"}' | gh workflow run triage.yml
+
+			# Run the workflow file 'triage.yml' with JSON via --json
+			$ gh workflow run triage.yml --json='{"name":"scully", "greeting":"hello"}'
+		`),
+		Args: func(cmd *cobra.Command, args []string) error {
+			if cmd.ArgsLenAtDash() == 0 && len(args[1:]) > 0 {
+				return cmdutil.FlagError{Err: fmt.Errorf("workflow argument required when passing input flags")}
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			if len(args) > 0 {
+				opts.Selector = args[0]
+				opts.InputArgs = args[1:]
+			} else if !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("workflow ID, name, or filename required when not running interactively")}
+			} else {
+				opts.Prompt = true
+			}
+
+			if !opts.IO.IsStdinTTY() {
+				if opts.JSON != "" {
+					return errors.New("JSON can only be passed on one of STDIN or --json at a time")
+				}
+				jsonIn, err := ioutil.ReadAll(opts.IO.In)
+				if err != nil {
+					return errors.New("failed to read from STDIN")
+				}
+				opts.JSON = string(jsonIn)
+			}
+
+			if opts.Selector == "" {
+				if opts.JSON != "" {
+					return &cmdutil.FlagError{Err: errors.New("workflow argument required when passing JSON")}
+				}
+			} else {
+				if opts.JSON != "" && len(opts.InputArgs) > 0 {
+					return &cmdutil.FlagError{Err: errors.New("only one of JSON or input arguments can be passed at a time")}
+				}
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runRun(opts)
+		},
+	}
+	cmd.Flags().StringVarP(&opts.Ref, "ref", "r", "", "The branch or tag name which contains the version of the workflow file you'd like to run")
+	cmd.Flags().StringVar(&opts.JSON, "json", "", "Provide workflow inputs as JSON")
+
+	return cmd
+}
+
+type InputAnswer struct {
+	providedInputs map[string]string
+}
+
+func (ia *InputAnswer) WriteAnswer(name string, value interface{}) error {
+	if s, ok := value.(string); ok {
+		ia.providedInputs[name] = s
+		return nil
+	}
+
+	// TODO i hate this; this is to make tests work:
+	if rv, ok := value.(reflect.Value); ok {
+		ia.providedInputs[name] = rv.String()
+		return nil
+	}
+
+	return fmt.Errorf("unexpected value type: %v", value)
+}
+
+func runRun(opts *RunOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not build http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("could not determine base repo: %w", err)
+	}
+
+	states := []shared.WorkflowState{shared.Active}
+	workflow, err := shared.ResolveWorkflow(
+		opts.IO, client, repo, opts.Prompt, opts.Selector, states)
```

Comment:
It feels slightly wrong to me that the user who hasn't specified the workflow ID will be presented with the list of workflows to choose from, but the listed workflows aren't guaranteed to be `workflow_dispatch` enabled. But, I find this acceptable since right now it would take N+1 queries to verify for each workflow if they have a workflow_dispatch trigger.